cmp/form/dropdown/ComboField.js

```diff
@@ -5,23 +5,29 @@
  * Copyright © 2018 Extremely Heavy Industries Inc.
  */
 
+import {PropTypes as PT} from 'prop-types';
 import {hoistComponent, elemFactory} from 'hoist/core';
 import {Classes, suggest} from 'hoist/kit/blueprint';
 
 import {BaseDropdownField} from './BaseDropdownField';
 
 /**
- * ComboBox Field
+ * ComboBox Field - A field with type ahead suggest and menu select
  *
- * @prop rest, see properties for HoistField
- *
- * @prop options, collection of form [{value: string, label: string}, ...] or [val, val, ...]
- * @prop placeholder, text to display when control is empty
- * @prop itemRenderer, optional custom itemRenderer, a function that receives (item, itemProps)
+ * See HoistField for properties additional to those documented below.
  */
 @hoistComponent()
 export class ComboField extends BaseDropdownField {
 
+    static propTypes = {
+        /** Collection of form [{value: string, label: string}, ...] or [val, val, ...] */
+        options: PT.arrayOf(PT.oneOfType([PT.object, PT.string])).isRequired,
+        /** Text to display when control is empty */
+        placeholder: PT.string,
+        /** Optional custom itemRenderer, a function that receives (item, itemProps) */
+        itemRenderer: PT.func
```

Comment:
Unclear to me what item / itemProps are in this context. (I have a TODO for myself to catch up on the various combobox-like-things APIs.)

Reply:
As I indicated in a comment below, optionsRender is clearer in terms of what this is. However, on reflection itemRenderer is what it is called in the blueprint docs, and itemProps is what gets passed by them. Perhaps a link to the blueprint docs is the better answer?

With regard to the propTypes for the mixed array. I couldn't find an example of this and it's not documented but I tested passing a mixed array of strings and objects and we had no trouble. Alternatively the docs state that you can pass a custom validation to arrayOf if we feel more comfortable with that.

Reply:
I would use `options` as we're taking over the API to some extent here, and using `options` for the data elements themselves (which I still think is a good idea).  Where we can provide the relevant info in a succinct form w/o linking to other docs, we should aim to do so (especially for a comp like this that will be very common).

Sounds like we are OK in terms of the array typing. I would hesitate to go further into a custom array validator at this point / for this use case.